app/pods/publications/route.js

```diff
@@ -1,6 +1,21 @@
 import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
 import Route from '@ember/routing/route';
+import { action } from '@ember/object';
+import { inject } from '@ember/service';
+import { hash } from 'rsvp';
 
 export default class PublicationsRoute extends Route.extend(AuthenticatedRouteMixin) {
+  @inject store;
 
+  @action
+  async model() {
```

Comment:
Merk op dat [de story](https://kanselarij.atlassian.net/browse/KAS-2436) specifieert van deze `findAll` in de [afterModel](https://api.emberjs.com/ember/3.17/classes/Route/methods/beforeModel?anchor=afterModel) hook uit te voeren.
`store` kan je als een globaal iets bekijken dat z'n state behoudt doorheen de applicatie (het is eigenlijk een `service` net zoals eender welke andere service). De verschillende `pulication-modes` die de store hier ophaalt, worden bewaard, en worden er later bij je `peekRecord` terug uit gehaald. Het is niet nodig `publicationModes` te returnen als deel van het model (en kan ook niet eens in de `afterModel`-hook.

Reply:
Ik snap het maar ik heb dit met Erika besproken. Mijn idee was dat het deel is van het model en dat alles in model() steken zorgt dat ge niet moet awaiten. Zij kon vond dit niet slecht.

Dan return ik het model om zeker te zijn dat alle routes sequentieel hun model opbouwen.

Reply:
Dan hebben we elkaar mis begrepen, denk ik. Je kan de data laden in de model-hook, maar returnen als deel van het model, zou ik ook niet doen, want de 'publication-modes' zijn niet het main model van deze route. Daarom volg ik eerder de redenering van @MikiDi om ze in de `afterModel` hook te laden.

De andere (sub)routes van publications zijn momenteel slechte voorbeelden qua model-hook. Ze maken bijna allemaal gebruik van `hash` en bevatten veel zaken die eigenlijk niet tot het model horen. Het gebruik van `hash` in de model-hook zou eerder een uitzondering dan de regel moeten zijn.

Reply:
OK, dit is nu veranderd. Al lijkt die `const model = hash(...)` toch beter. Dan kunnen alle requests tegelijk runnen en zijn ze meteen beschikbaar in de controller/template.